src/main/kotlin/logic/usecase/GetCityLocationByNameUseCase.kt

```
@@ -0,0 +1,16 @@
+package logic.usecase
+
+import logic.model.CityLocation
+import logic.repository.WeatherRepository
+
+class GetCityLocationByNameUseCase(
+    private val weatherRepository: WeatherRepository
+) {
+    suspend fun execute(cityName: String): CityLocation {
+        if (cityName.isBlank()) {
```

you can use built in validation function that called require(condition) { "write exception msg hrer"} 

this built in function throws IllegalArgumentException if condition is false.

---

I know it, but I guess if you need to throw a custom exception, it won't help you.

---

As Bareq mentioned, it's recommended to use more descriptive names for your use cases, rather than generic ones like **execute** or **invoke**